packages/core/src/discovery.js

```diff
@@ -346,7 +351,8 @@ export function createDiscoveryQueue(percy) {
             yield* captureSnapshotResources(page, snapshot, {
               captureWidths: !snapshot.domSnapshot && percy.deferUploads,
               capture: callback,
-              captureForDevices: percy.deviceDetails || []
+              captureForDevices: percy.deviceDetails || [],
+              captureResponsiveAssetsEnabled: !!percy.deviceDetails
```

Comment:
Not sure why this is required. On line 166 wouldnt it be always true ? if array is empty it will never go in loop